### ISSUE TITLE
Using  sub-command always invokes * 

### ISSUE THREAD

**Issue by soichih** [tj/commander.js#809]
With following program

```
  
program
  .command('com1', 'for test-com1.js')
  .command('com2', 'for test-com2.js')

program
  .command('*')
  .action(env=>{
      console.log("unknown command");
  })

program.parse(process.argv);

```

'*' will always be invoked regardless of what command user tries to execute.

```
hayashis@dev1:~/tmp/test $ node test.js com1
unknown command
here is test-com1.js
hayashis@dev1:~/tmp/test $ node test.js com2
unknown command
here is test-com2.js
hayashis@dev1:~/tmp/test $ node test.js bogus
unknown command

```

I believe "unknown command" should be displayed only when user enter non-existing command name.

**Comment by linfan68** [comment]
Hit this issue yesterday...
And I found it is related the help string passed as the second parameter of .command()
Removing the help string fixed it for me

Also I found there are more issues, like if you’re using command with parameter ‘cmd <parameter>’, adding help string make it not checking the parameter anymore, and parameter parsing also seems wrong (it pass in the command name string instead)

**Comment by JohnRGee** [comment]
I agree Soichi, this looks like a shortcoming with handling of mixed git-style commands and commands with actions.

A work-around might be to do the error/help detection in a git-style command too:

````
program
  .command('com1', 'for test-com1.js')
  .command('com2', 'for test-com2.js')
  .command('unknown-command', 'for test-unknown-command.js', { isDefault:true })

program.parse(process.argv);
````

**Comment by JohnRGee** [comment]
FYI @linfan68 when `.command()` is invoked with a description argument, this tells commander that you're going to use separate executables for sub-commands. See [Git-style sub-commands](https://github.com/tj/commander.js#git-style-sub-commands)

**Comment by shadowspawn** [comment]
`command:*` is not emitted for executable subcommands in the published pre-release of Commander v5.0.0 (#1163)

On a related note, by default unknown commands will display an error (so you may not need to listen for `command:*` yourself.)

**Comment by shadowspawn** [comment]
Commander v5.0.0 has been released.

https://github.com/tj/commander.js/releases/tag/v5.0.0
